jupyterhub/user.py

```diff
@@ -622,6 +622,19 @@ async def spawn(self, server_name='', options=None, handler=None):
             if callable(allowed_roles):
                 allowed_roles = allowed_roles(spawner)
 
+            # allowed_roles config is a list of strings
+            # oauth provider.allowed_roles is a list or orm.Roles
```

Comment:
```suggestion
            # oauth provider.allowed_roles is a list of orm.Roles
```

Reply:
Thanks, fixed! Test should pass after rebasing after merge of #3719